apps/src/blockly/addons/functionBlocks.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Used to customize function definition blocks for Google Blockly labs.
+ * Derived from core Google Blockly:
+ * https://github.com/google/blockly/blob/5a23c84e6ef9c0b2bbd503ad9f58fa86db1232a8/blocks/procedures.js#L449-L490
+ */
+export const FUNCTION_BLOCK = {
+  init: function() {
+    const initName = Blockly.Procedures.findLegalName('', this);
+    const nameField = new Blockly.FieldTextInput(
+      initName,
+      Blockly.Procedures.rename
+    );
+    nameField.setSpellcheck(false);
+    this.appendDummyInput()
+      .appendField(' ')
+      .appendField(nameField, 'NAME')
+      .appendField('', 'PARAMS');
+    this.setStyle('procedure_blocks');
+    this.arguments_ = [];
+    this.argumentVarModels_ = [];
+    this.setStatements_(true);
+    this.statementConnection_ = null;
+  }
+};
+
+/**
+ * Constructs the blocks required by the flyout for the procedure category.
+ * Modeled after core Blockly procedures flyout category, but excludes unwanted blocks.
+ * Derived from core Google Blockly:
+ * https://github.com/google/blockly/blob/5a23c84e6ef9c0b2bbd503ad9f58fa86db1232a8/core/procedures.ts#L202-L287
+ * @param {WorkspaceSvg} workspace The workspace containing procedures.
+ * @returns XML block elements
+ */
+export function FUNCTION_CATEGORY(workspace) {
+  const functionDefinitionBlock = newDefinitionBlock();
+
+  // Find all user-created procedure definitions in the workspace.
+  // allProcedures returns a pair of arrays, but we only need the first.
+  // The second contains procedures with return variables which we don't support.
+  // Each procedure is defined by a three-element list of name, parameter list,
+  // and return value boolean (false).
+  // https://developers.google.com/blockly/reference/js/blockly.procedures_namespace.allprocedures_1_function.md
+  const allWorkspaceProcedures = Blockly.Procedures.allProcedures(workspace)[0];
+  const functionCallBlocks = allCallBlocks(allWorkspaceProcedures);
+  return functionDefinitionBlock.concat(functionCallBlocks);
+}
+
+export function newDefinitionBlock() {
```

Comment:
Nit: name implies to me that a block is returned -- could we return the block, and then wrap it in an array where its called?